daemon/endpoint.go

```diff
@@ -1051,3 +1056,27 @@ func (e *endpointManager) DeleteEndpoint(ep *endpoint.Endpoint) int {
 func (e *endpointManager) deleteEndpointQuiet(ep *endpoint.Endpoint, conf endpoint.DeleteConfig) []error {
 	return ep.Delete(e.d, e.d.ipam, e, conf)
 }
+
+// Check verifies that endpoints are alive and healthy.
+func (e *endpointManager) Check(ep *endpoint.Endpoint) error {
+	// This only lives in endpointManager for convenience, the real
+	// intended home would be likely be pkg/datapath somewhere that
+	// understands how to gauge endpoint health from a node perspective.
+
+	// Be extra careful, we're only looking for one specific type of error
+	// currently: That the link has gone missing. Ignore other error to
+	// ensure that the caller doesn't unintentionally tear down the
+	// Endpoint thinking that it no longer exists.
+	_, err := netlink.LinkByName(ep.InterfaceName())
```

Comment:
just to be sure, this can be done from the Cilium's container?

Reply:
Yes, the Cilium container manages eBPF programs attached to these devices so it must inherently have access to this.